packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -155,8 +155,14 @@ function Tooltip<T extends React.ElementType>({
   }
 
   function onMouseEnter() {
-    setIsPointerIntersecting(true);
-    setOpen(true, enterDelayMs);
+    // Interactive Tags should not support onMouseEnter
+    const isMouseOnEnterDisabled = customClassName?.includes(
+      `${prefix}--tooltip-disabled-onmouseenter`
+    );
```

Comment:
Is there a different way this could be accomplished? Using a classname to determine this feels super brittle and worries me.

More broadly, this feels like it erodes at the specced functionality of a Tooltip. imo all Tooltips should behave uniformly. What's the reason for this to not follow the established pattern of supporting onMouseEnter?

If there's a use case established here for disabling onMouseEnter, I wonder if we should document that reasoning and extend the option to consumers via a new prop?

Reply:
There is a different way that would be using the `enterDelayMs` from `Tooltip`. We could set a high value to avoid the activation. Also we could add a prop on `Tooltip`, but I'm against that because of your second paragraph, we don't want users to remove a core functionality from a `Tooltip`.

That was something @mbgower and @laurenmrice disccused, and the reason for removing the `onMouseEnter` is that we don't want users to think that could be interactive by mouse. Correct me if I didn't understand correctly Lauren/Gower.

Reply:
The problem we are trying to solve is that if a Tag, in any variant (Read-only, Dismissible, Selectable, Operational), has a tag title too long for its container width, it should truncate in the tag, and the full title should be able to be revealed to the user.

We decided to expose the full name of the truncated tag title in a tooltip if the user focuses on the tag. The tricky part is that we have a Read-only tag variant, which is not supposed to be "interactive". We felt this way was acceptable enough to apply to the Read-only tag, as with the other tag variants, because it has limited interactive functionality but still solves the problem. I also feel that truncated tag titles probably does not happen often, we just don't want them to wrap to a second line in the tag. 

If there is a better approach to solving the above, we can change it! But this is the original solution we had come up with. We discussed this during a design critique and a CAG call in February, but would be happy to discuss it again.

Reply:
One other thing to consider is the click action on a `Tag`. When a `Tag` is clicked it receives focus for a brief moment and the `Tooltip` is activated which can cause a weird flicker looking like a "bug". Having the `onMouseEnter` as it is would avoid that.

Reply:
Thanks for the context! @guidari what if we provided an undocumented way to arrest the default functionality inside `onMouseEnter`? Something vaguely like this:

```js
  function onMouseEnter() {
    if (!...rest?.onMouseEnter?.()) {
      setIsPointerIntersecting(true);
      setOpen(true, enterDelayMs);
    }
  }
  
// Then within the tag we can return false
<Tooltip
  ...
  onMouseEnter={() => false}
>
  
```

This is still a bit wonky, but at least provides direct control using the existing API without relying on a classname that can change or be removed, etc.

We're already not allowing `onMouseEnter` to be user-configurable (it's never pulled off `rest`, so I think this wouldn't be changing anything a consumer would be using.